client/src/store.js

```diff
@@ -8,7 +8,7 @@ Vue.use(Vuex);
 export default new Vuex.Store({
   state: {
     location: null,
-    pipelines: [],
+    pipelines: {},
```

Comment:
Leaving the array `[]` was a mistake.  What do you think about making this `null` by default?  The check for emptiness is nicer: `pipelines === null`  rather than enumerating its key list.